contracts/escrows/CountdownGriefingEscrow.sol

```diff
@@ -0,0 +1,357 @@
+        address seller,
+        address operator,
+        uint256 paymentAmount,
+        uint256 stakeAmount,
+        uint256 escrowCountdown,
+        bytes memory metadata,
+        bytes memory agreementParams
+    ) public initializeTemplate() {
+        // set participants if defined
+        if (buyer != address(0))
+            _data.buyer = buyer;
+        if (seller != address(0))
+            _data.seller = seller;
+
+        // set operator
+        if (operator != address(0)) {
+            Operated._setOperator(operator);
+            Operated._activateOperator();
+        }
+
+        // set amounts if defined
+        if (paymentAmount != uint256(0))
+            require(paymentAmount <= uint256(uint128(paymentAmount)), "paymentAmount is too large");
+            _data.paymentAmount = uint128(paymentAmount);
+        if (stakeAmount != uint256(0))
+            require(stakeAmount == uint256(uint128(stakeAmount)), "stakeAmount is too large");
+            _data.stakeAmount = uint128(stakeAmount);
+
+        // set countdown length
+        Countdown._setLength(escrowCountdown);
+
+        // set metadata
+        if (metadata.length != 0)
+            EventMetadata._setMetadata(metadata);
+
+        // set agreementParams
+        if (agreementParams.length != 0) {
+            (
+                uint120 ratio,
+                uint8 ratioType,
+                uint128 agreementCountdown
+            ) = abi.decode(agreementParams, (uint120, uint8, uint128));
+
+            _data.agreementParams = AgreementParams(ratio, ratioType, agreementCountdown);
+        }
+
+        // emit event
+        emit Initialized(buyer, seller, operator, paymentAmount, stakeAmount, escrowCountdown, metadata, agreementParams);
+    }
+
+    /// @notice Emit metadata event
+    /// @dev Access Control: seller OR buyer OR operator
+    ///      State Machine: always
+    function setMetadata(bytes memory metadata) public {
+        // restrict access
+        require(isSeller(msg.sender) || isBuyer(msg.sender) || Operated.isActiveOperator(msg.sender), "only seller or buyer or active operator");
+
+        // update metadata
+        EventMetadata._setMetadata(metadata);
+    }
+
+    /// @notice Deposit Stake
+    ///          - if seller not already set, make msg.sender the seller
+    ///          - if buyer already deposited the payment, finalize the escrow
+    /// @dev Access Control: buyer OR operator
+    ///      State Machine: before finalize() OR before cancel()
+    function depositStake() public {
+        // restrict access control
+        // set msg.sender as seller if not already set
+        if (!isSeller(msg.sender) && !Operated.isActiveOperator(msg.sender)) {
+            require(_data.seller == address(0), "only seller or active operator");
+            _data.seller = msg.sender;
+        }
+        // restrict state machine
+        require(!isFinalized() && !isCancelled(), "only before finalize or cancel");
```

Comment:
Since the seller can cancel at anytime, they could frontrun and block this call from ever succeeding. I'm not sure why someone would do that in practice, but it's a definite possibility.

Reply:
Frontrunning is acceptable